### PR TITLE
Chm gem subgrid

### PR DESCRIPTION
**Pull request by VVionnet**
Merge to include several minor options and bug fixes in CHM:
- Correction of snow compaction in presence of liquid water in Snobal

- New option to correct vertically precipitation using vertical lapse rate derived from stations at Marmot Creek

- New option in the snow slide routine to compute maximum holding capacity as a function of vertical snow depth

### CODE REVIEW

File: src/modules/interp_met/WindNinja.cpp [hunk 1]
```diff
@@ -241,15 +241,20 @@ void WindNinja::run(mesh& domain)
...
                W_transf = 1.+(Max_spdup-1.)*(W_transf-1.)/(transf_max-1.);
 
-           if (ninja_recirc){  // Need further test
+           if(compute_Sx){
+               if (ninja_recirc){  // Need further test
  
               //Compute what liston calls 'wind slope' using updated wind direction
-              double omega_s = face->slope() * cos(theta - face->aspect());
-              (*face)["omega_s"_s]= omega_s;
+            //  double omega_s = face->slope() * cos(theta - face->aspect());
```
Comment (Chrismarsh) [inline src/modules/interp_met/WindNinja.cpp]: Can these comments be removed?

File: src/modules/interp_met/p_lapse.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+    LOG_DEBUG << "Successfully instantiated module " << this->ID;
+}
+void p_lapse::init(mesh& domain)
+{
+    ompException oe;
+#pragma omp parallel for
+    for (size_t i = 0; i < domain->size_faces(); i++)
+    {
+      oe.Run([&]
```
Comment (Chrismarsh) [inline src/modules/interp_met/p_lapse.cpp]: remove the oe.Run stuff. Kevin and I found a pretty large performance regression using it in a tight loop like this. I need to come up with a better solution

File: src/modules/interp_met/p_lapse.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    {
+        mf /= 100.0; //to m^-1
+    }
+    std::vector< boost::tuple<double, double, double> > ppt;
+    std::vector< boost::tuple<double, double, double> > staion_z;
+    for (auto& s : global_param->get_stations( face->get_x(), face->get_y()))
+    {
+        if( is_nan(s->get("p")))
+            continue;
+        double u = s->get("p");
```
Comment (Chrismarsh) [inline src/modules/interp_met/p_lapse.cpp]: u should be p or ?

Comment (VVionnet) [reply src/modules/interp_met/p_lapse.cpp]: No it is correct. See the same lines in Thornton_p.cpp

File: src/modules/interp_met/p_lapse.cpp (hunk 2 above)
Comment (VVionnet) [reply src/modules/interp_met/p_lapse.cpp]: OK. Done

File: src/modules/interp_met/WindNinja.cpp (hunk 1 above)
Comment (VVionnet) [reply src/modules/interp_met/WindNinja.cpp]: I would prefer to keep it here as another option to correct wind on leeward slopes. Everything is really an early stage development

**Pull request merged**